### PR TITLE
Taylor - Manager Single Community

### PR DESCRIPTION
**Pull request by TaylorDJones11**
## What type of PR is this? (check all applicable)

- [ ] Feature
- [ ] Refactor
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Other (please specify here: Manager Single Community Component)
  
  
## Description of what this Pull Request does
- Currently no functionality with buttons but will later add - need to discuss if the buttons will take user to separate page or using modals like in wireframes.
- The Manager Single Community(MSC) will allow users to edit, view and delete their community page.
- The MSC will display the community photo, community name, and community status.

## Related Tickets & Documents
- Closes: https://github.com/AritDeveloperCircle/findyourpeople/issues/50
- Related Issues: https://github.com/AritDeveloperCircle/findyourpeople/issues/51
  
## QA Instructions, Screenshots, Recordings
Screenshot 2023-05-03 at 20 31 47

  
## Added/updated tests?
- [ ] Yes
- [ ] Not Yet, and here is why: __explain here__
- [ ] I need help with tests
- [x] No, and here is why: app is currently not being tested

### CODE REVIEW

**Review by jjleigh — APPROVED** [review]
Looks good

File: src/components/managersinglecomm/ManagerSingleCommunity.js
Comment (msarit) [inline src/components/managersinglecomm/ManagerSingleCommunity.js:8]: Hey @TaylorDJones11 so we decided that using `camelCase` in the CSS files is actually recommended for React. 😅 So please return to using `camelCase` for your CSS properties, as this will be our convention moving forward. 🙏🏾

Comment (msarit) [reply src/components/managersinglecomm/ManagerSingleCommunity.js:8]: Everything else looks great; I will approve after CSS case changes

**Review by msarit — CHANGES_REQUESTED** [review]

**Review by msarit — CHANGES_REQUESTED** [review]
Great effort @TaylorDJones11 

First off, I think the word `Listing` is better for naming this component, so that we have the easy `listings` plural word. We already have a `<Listing>` component created by Ibimina (for the landing page), so yours can be called `DashboardListing`.

Secondly, I am thinking that it would make sense to include the parent component for this `<DashboardListing>` component, in this PR.

For example, a component called `<DashboardListings>`:

```js
import DashboardListing from "@/path/to/file"
import styles from "@/path/to/css.file"

// function that fetches listings from Firebase
const listings = function(...);

function DashboardListings({ listings )) {
   return (
      <section className={styles.dashboard_container}>
          {listings.map((listing) => (
            <DashboardListing data={listing} key={listing.id} />
          ))}
       </section>
    )
}

export default DashboardListings;
```

File: src/components/managersinglecomm/ManagerSingleCommunity.js
Comment (TaylorDJones11) [reply src/components/managersinglecomm/ManagerSingleCommunity.js:1]: > @TaylorDJones11 I think instead of `managersinglecomm` in the file path, we can use the simpler `dashboard` what do you think?

I was only doing the Single Community Component while @istoga has the entire dashboard. I can change the name but I didn't want to get it mixed up with Neka's possibly name

File: src/pages/index.js
Comment (TaylorDJones11) [reply src/pages/index.js]: > Sigh .. we are in trouble oh! With these single vs double quotes linters. Please revert back to the double quotes, else we will never make progress. That's the reason for this file's entire diff - your linter has forced the use of single quotes in this file.

Yes, I had my VS settings to single quotes but changed it back to double. Should I push to the file again with double quotes?

**Comment by TaylorDJones11** [comment]
> Great effort @TaylorDJones11
> 
> First off, I think the word `Listing` is better for naming this component, so that we have the easy `listings` plural word. We already have a `<Listing>` component created by Ibimina (for the landing page), so yours can be called `DashboardListing`.
> 
> Secondly, I am thinking that it would make sense to include the parent component for this `<DashboardListing>` component, in this PR.
> 
> For example, a component called `<DashboardListings>`:
> 
> ```js
> import DashboardListing from "@/path/to/file"
> import styles from "@/path/to/css.file"
> 
> // function that fetches listings from Firebase
> const listings = function(...);
> 
> function DashboardListings({ listings )) {
>    return (
>       <section className={styles.dashboard_container}>
>           {listings.map((listing) => (
>             <DashboardListing data={listing} key={listing.id} />
>           ))}
>        </section>
>     )
> }
> 
> export default DashboardListings;
> ```

A bit confused here. 
1. I should change the name from "ManagerSingleCommunity" to "DashboardListing" ?
2. I think this is where Neka's ticket would take over?

**Comment by msarit** [comment]
> * I think this is where Neka's ticket would take over?

Pls link Neka's PR

**Comment by TaylorDJones11** [comment]
> > * I think this is where Neka's ticket would take over?
> 
> 
> 
> Pls link Neka's PR

I linked it above when I first wrote the PR. Should it be linked twice?

**Review by istoga — COMMENTED** [review]
I added the component into the landing page it seemed a bit stretched out, but nothing a bit of CSS cannot fix. Looks great @TaylorDJones11 ! Thanks for doing this, will be on the lookout for this to get merged.

File: src/components/managersinglecomm/ManagerSingleCommunity.js
Comment (msarit) [reply src/components/managersinglecomm/ManagerSingleCommunity.js]: This is alright; we can fix any clashes during merge conflict resolution.
I think it makes sense for all `dashboard`-related components to live inside `dashboard` - what do you think @istoga?

File: src/pages/index.js
Comment (msarit) [reply src/pages/index.js]: Yes please do!

**Comment by msarit** [comment]
> > Great effort @TaylorDJones11
> > First off, I think the word `Listing` is better for naming this component, so that we have the easy `listings` plural word. We already have a `<Listing>` component created by Ibimina (for the landing page), so yours can be called `DashboardListing`.
> > Secondly, I am thinking that it would make sense to include the parent component for this `<DashboardListing>` component, in this PR.
> > For example, a component called `<DashboardListings>`:
> > ```js
> > import DashboardListing from "@/path/to/file"
> > import styles from "@/path/to/css.file"
> > 
> > // function that fetches listings from Firebase
> > const listings = function(...);
> > 
> > function DashboardListings({ listings )) {
> >    return (
> >       <section className={styles.dashboard_container}>
> >           {listings.map((listing) => (
> >             <DashboardListing data={listing} key={listing.id} />
> >           ))}
> >        </section>
> >     )
> > }
> > 
> > export default DashboardListings;
> > ```
> 
> A bit confused here.
> 
> 1. I should change the name from "ManagerSingleCommunity" to "DashboardListing" ?
> 2. I think this is where Neka's ticket would take over?

I understand your concerns. Since @istoga is building the actual container that will hold the individual Dashboard Listings (although I truly feel that these two tasks could be knocked out in one PR - especially considering our remaining time, but I defer to how you want to run things 😅 ), you may just rename your component to `<DashboardListing>`. Thanks!

File: src/components/managersinglecomm/ManagerSingleCommunity.js
Comment (TaylorDJones11) [reply src/components/managersinglecomm/ManagerSingleCommunity.js]: > This is alright; we can fix any clashes during merge conflict resolution. I think it makes sense for all `dashboard`-related components to live inside `dashboard` - what do you think @istoga?

I've changed the component folders/function/styling to match with "Dashboard" throughout

File: src/components/managersinglecomm/managersinglecomm.module.css
Comment (TaylorDJones11) [reply src/components/managersinglecomm/managersinglecomm.module.css]: > Not sure why we are mixing `rem` and `px` units? Rationale?

Certain units help control spacing a bit better for me. I've looked at other CSS modules in the repo, some are using mixed units, so I didn't think it would be an issue to do so. I can ask the group which units we'd like to stick to when starting to design because I've seen 'rem' 'em' and 'px'.

**Comment by msarit** [comment]
> Currently no functionality with buttons but will later add - need to discuss if the buttons will take user to separate page or using modals like in wireframes.

@TaylorDJones11 We shall have a separate page for the Listing Edits. I think it will be easier. Thoughts? @annonymousauthority @jjleigh?

**Review by msarit — APPROVED** [review]

**Pull request merged**
